### PR TITLE
feat(es/minifier): Inline more with sequential inliner

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

I wrote descriptions in the code with comments.

**Related issue:**

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/single-pass/pr-6127/1/output.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+console.log((function() {
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/single-pass/pr-6127/1/output.js:1]: Previous input:

```js
console.log((function() {
    var b = 0;
    return b |= 94, b <<= 8, b |= 173, b <<= 8, b |= 190, b <<= 8, b |= 239;
})().toString(16));
```

Comment (Austaras) [reply crates/swc_ecma_minifier/tests/single-pass/pr-6127/1/output.js:1]: What about update expr? Something like
```js
var a = 10;
a++,++a,a--,--a
console.log(10)
```

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/single-pass/pr-6127/1/output.js:1]: It should be handled in future IMO

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/projects/output/react-dom-17.0.2.js [hunk 2]
```diff
@@ -7847,7 +7847,7 @@
     }
     function unbatchedUpdates(fn, a) {
         var prevExecutionContext = executionContext;
-        executionContext &= -2, executionContext |= 8;
+        executionContext = -2;
```
[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_minifier/tests/projects/output/react-dom-17.0.2.js:7850]: I think you are right

File: crates/swc_ecma_minifier/tests/benches-full/terser.js [hunk 3]
```diff
@@ -1924,8 +1924,8 @@
         base && base.PROPS && (props = props.concat(base.PROPS));
         for(var code = "return function AST_" + type + "(props){ if (props) { ", i = props.length; --i >= 0;)code += "this." + props[i] + " = props." + props[i] + ";";
         const proto = base && Object.create(base.prototype);
-        (proto && proto.initialize || methods && methods.initialize) && (code += "this.initialize();"), code += "}", code += "this.flags = 0;";
-        var ctor = Function(code += "}")();
+        (proto && proto.initialize || methods && methods.initialize) && (code += "this.initialize();");
+        var ctor = Function(code = (code += "}") + "this.flags = 0;}")();
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/benches-full/terser.js:1928]: I think we can optimize this further

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/benches-full/terser.js:1928]: I tried and concluded that it would be better to do this with another PR

**Review by kdy1 — COMMENTED** [review]
This pr is now correct, but this increases the size in general.
I'll try handling only literals

File: crates/swc_ecma_minifier/tests/pass-default/pr-6127/1/output.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+console.log((function() {
+    var b = 0;
+    return b |= 94, b <<= 8, b |= 173, b <<= 8, b |= 190, b <<= 8, b |= 239;
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/pass-default/pr-6127/1/output.js:3]: This is not ideal

**Comment by kdy1** [comment]
This was a wrong approach

**Pull request closed without merging**